plugins/woocommerce/includes/data-stores/abstract-wc-order-item-type-data-store.php

```diff
@@ -163,4 +215,33 @@ public function clear_cache( &$item ) {
 		wp_cache_delete( 'order-items-' . $item->get_order_id(), 'orders' );
 		wp_cache_delete( $item->get_id(), $this->meta_type . '_meta' );
 	}
+
+	/**
+	 * Persist the Cost of Goods Sold related data to the database.
+	 *
+	 * @param WC_Order_Item $item The order item for which the data will be persisted.
+	 */
+	private function save_cogs_data( WC_Order_Item $item ) {
+		$cogs_value = $item->get_cogs_value();
+
+		/**
+		 * Filter to customize the Cost of Goods Sold value that gets saved for a given order item,
+		 * or to suppress the saving of the value (so that custom storage can be used).
+		 *
+		 * @since 9.5.0
+		 *
+		 * @param float|null $cogs_value The value to be written to the database. If returned as null, nothing will be written.
+		 * @param WC_Order_Item $item The order item for which the value is being saved.
+		 */
+		$cogs_value = apply_filters( 'woocommerce_save_order_item_cogs_value', $cogs_value, $item );
+		if ( is_null( $cogs_value ) ) {
+			return;
+		}
+
+		if ( 0.0 === $cogs_value ) {
+			$this->order_item_data_store->delete_metadata( $item->get_id(), '_cogs_value', '', true );
+		} else {
+			$this->order_item_data_store->update_metadata( $item->get_id(), '_cogs_value', $cogs_value );
```

Comment:
Since the value is filtered, could it be worth casting `$cogs_value` to a float at this point, or else testing to ensure it is a float/that it is castable upon retrieval?

Reply:
Yes, good point, things like this have bitten us in the past.